vllm_ascend/ops/rotary_embedding.py

```diff
@@ -214,21 +221,49 @@ def _set_cos_sin_cache(self, seq_len, device, dtype):
     t = torch.arange(seq_len, device=device, dtype=torch.float32)
 
     freqs = torch.outer(t, inv_freq)
+    cache = torch.cat([freqs.cos() * self.mscale,
+                       freqs.sin() * self.mscale],
+                      dim=-1).to(dtype)
+    self.register_buffer("cos_sin_cache", cache, persistent=False)
 
-    # _mscale = float(
-    #     yarn_get_mscale(self.scaling_factor, self.mscale)
-    #     / yarn_get_mscale(self.scaling_factor, self.mscale_all_dim)
-    # )
 
-    emb = torch.cat((freqs, freqs), dim=-1)
-    self.register_buffer("cos_cached", (emb.cos() * self.mscale).to(dtype),
-                         persistent=False)
-    self.register_buffer("sin_cached", (emb.sin() * self.mscale).to(dtype),
-                         persistent=False)
+def deepseek_rope_init_func(
+    self,
+    head_size: int,
+    rotary_dim: int,
+    max_position_embeddings: int,
+    base: int,
+    is_neox_style: bool,
+    scaling_factor: float,
+    dtype: torch.dtype,
+    *,
+    extrapolation_factor: float = 1,
+    attn_factor: float = 1,
+    beta_fast: int = 32,
+    beta_slow: int = 1,
+    mscale: float = 1,
+    mscale_all_dim: float = 0,
+) -> None:
+    self.scaling_factor = scaling_factor
+    self.extrapolation_factor = extrapolation_factor
+    self.attn_factor = attn_factor
+    self.beta_fast = beta_fast
+    self.beta_slow = beta_slow
+    # Get n-d magnitude scaling corrected for interpolation.
+    self.mscale = float(
+        yarn_get_mscale(self.scaling_factor, float(mscale)) /
+        yarn_get_mscale(self.scaling_factor, float(mscale_all_dim)) *
+        attn_factor)
+    super(DeepseekScalingRotaryEmbedding,
+          self).__init__(head_size, rotary_dim, max_position_embeddings, base,
+                         is_neox_style, dtype)
+    self.max_seq_len = max_position_embeddings
+    _set_cos_sin_cache(self,
+                       max_position_embeddings,
+                       dtype=dtype,
+                       device="npu")
 
 
-# TODO: Patch when aclnn ops avaiable
 RotaryEmbedding.forward_oot = rope_forward_oot
+DeepseekScalingRotaryEmbedding.__init__ = deepseek_rope_init_func
```

Comment:
For this kind of change which override vllm. We should add note to describe why. L269 is the same.

Reply:
the DeepseekScalingRotaryEmbeeding is not a custom op in stock vLLM? Can we file a PR to vLLM to make it a custom op, after that can use the forward_oot, right?